dbos/system_database.py

```diff
@@ -69,6 +58,9 @@ class WorkflowStatusInternal(TypedDict):
     app_id: Optional[str]
     request: Optional[str]  # JSON (jsonpickle)
     recovery_attempts: Optional[int]
+    authenticated_user: Optional[str]
+    assumed_role: Optional[str]
+    authenticated_roles: Optional[str]  # Comma-delimited list of roles.
```

Comment:
To be consistent with TypsScript and searchable via PG json type, can we use serialized JSON here?